src/laser_geometry.cpp

```diff
@@ -427,8 +427,8 @@ void LaserProjection::transformLaserScanToPointCloud_(
   TIME end_time = scan_in.header.stamp;
   // TODO(anonymous): reconcile all the different time constructs
   if (!scan_in.ranges.empty()) {
-    end_time = end_time + rclcpp::Duration(
-      static_cast<int>((scan_in.ranges.size() - 1) * scan_in.time_increment), 0);
+    end_time = start_time + rclcpp::Duration::from_seconds(
+      (scan_in.ranges.size() - 1) * scan_in.time_increment);
```

Comment:
My first thought was that it would be better to convert `scan_in.time_increment` to an `rclcpp::Duration` first and then multiply by `(scan_in.ranges.size() - 1)`. (Because the Duration should be smart about handling operations on it in ways that preserve precision.) But it looks like `rclcpp::Duration` doesn't overload the multiplication operator.

Nitpick: Technically there is a loss of precision here because `scan_in.time_increment` is a 32 bit float and you're multiplying it  by a 64 bit unsigned integer which makes it a larger value, potentially making the exponent of the float bigger and losing bits in the mantissa.  The alternative would be to cast `scan_in.time_increment` to a double before multiplying. (`rclcpp::Duration::from_seconds()` takes a double anyway). What do you think about:
```c++
end_time = start_time + rclcpp::Duration::from_seconds(
      static_cast<double>(scan_in.ranges.size() - 1) * static_cast<double>(scan_in.time_increment));
```

Reply:
Hi, thanks for the review.

I like this explicit casting better as well. I will update the PR accordingly :)

Reply:
:+1: